builtin/credential/aws/path_role.go

```diff
@@ -320,6 +322,64 @@ func (b *backend) setRole(ctx context.Context, s logical.Storage, roleName strin
 	return nil
 }
 
+// updateUpgradableRoleEntries upgrades and persists all of the role entries
+// that are in need of being upgraded.
+func (b *backend) updateUpgradableRoleEntries(ctx context.Context, req *logical.InitializationRequest) error {
+
+	// Upgrade only if we are either: (1) a local mount, or (2) are _not_ a
+	// performance replicated standby cluster.
+	if !(b.System().LocalMount() || !b.System().ReplicationState().HasState(
```

Comment:
This will actually do the opposite of what you want  -- in these cases we want to continue on, not return nil.

Reply:
It was actually correct, in that there was a `!` wrapping the whole predicate, but I agree that that is confusing, so I just pushed a commit that simplifies the logic

Reply:
Ah, right. OK, thanks.